src/txadmission.cpp

```diff
@@ -0,0 +1,1028 @@
+            extraFlags |= SCRIPT_ENABLE_MAY152018_OPCODES;
+        }
+
+        // Check against previous transactions
+        // This is done last to help prevent CPU exhaustion denial-of-service attacks.
+        unsigned char sighashType = 0;
+        if (!CheckInputs(*tx, state, view, true, STANDARD_SCRIPT_VERIFY_FLAGS | extraFlags, true, &resourceTracker,
+                nullptr, &sighashType))
+        {
+            LOG(MEMPOOL, "CheckInputs failed for tx: %s\n", tx->GetHash().ToString().c_str());
+            return false;
+        }
+        entry.UpdateRuntimeSigOps(resourceTracker.GetSigOps(), resourceTracker.GetSighashBytes());
+
+#if 0 // to be removed -- bugs in this should have been shaken out
+        // Check again against just the consensus-critical mandatory script
+        // verification flags, in case of bugs in the standard flags that cause
+        // transactions to pass as valid when they're actually invalid. For
+        // instance the STRICTENC flag was incorrectly allowing certain
+        // CHECKSIG NOT scripts to pass, even though they were invalid.
+        //
+        // There is a similar check in CreateNewBlock() to prevent creating
+        // invalid blocks, however allowing such transactions into the mempool
+        // can be exploited as a DoS attack.
+        unsigned char sighashType2 = 0;
+        if (!CheckInputs(*tx, state, view, true, MANDATORY_SCRIPT_VERIFY_FLAGS | extraFlags, true, nullptr, nullptr,
+                &sighashType2))
+        {
+            return error(
+                "%s: BUG! PLEASE REPORT THIS! ConnectInputs failed against MANDATORY but not STANDARD flags %s, %s",
+                __func__, hash.ToString(), FormatStateMessage(state));
+        }
+
+        entry.sighashType = sighashType | sighashType2;
+#endif
+
+        entry.sighashType = sighashType;
+        // This code denies old style tx from entering the mempool as soon as we fork
+        if (IsUAHFforkActiveOnNextBlock(chainActive.Tip()->nHeight) && !IsTxUAHFOnly(entry))
+        {
+            return state.Invalid(false, REJECT_WRONG_FORK, "txn-uses-old-sighash-algorithm");
+        }
+
+        respend.SetValid(true);
+        if (respend.IsRespend())
+        {
+            return state.Invalid(false, REJECT_CONFLICT, "txn-mempool-conflict");
+        }
+
+        {
+            READLOCK(pool.cs);
+            CTxMemPool::setEntries setAncestors;
+            // note we could resolve ancestors to hashes and return those if that saves time in the txc thread
+            if (!pool._CalculateMemPoolAncestors(entry, setAncestors, nLimitAncestors, nLimitAncestorSize,
+                    nLimitDescendants, nLimitDescendantSize, errString))
+            {
+                return state.DoS(0, false, REJECT_NONSTANDARD, "too-long-mempool-chain", false, errString);
+            }
+        }
+
+        {
+            boost::unique_lock<boost::mutex> lock(csCommitQ);
+            txCommitQ[eData.hash] = eData;
```

Comment:
- boost::unique_lock<boost::mutex> lock(csCommitQ);
+ std::unique_lock<std::mutex> lock(csCommitQ);

Reply:
all of sync.h is using boost::mutex and boost::unique_locks, which means that almost all of our code is using the boost variants.  I think that in a different PR we should move these over all together.